src/models/room-state.js

```diff
@@ -247,6 +247,47 @@ RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
 };
 
 /**
+ * Returns true if userId is in room, event is not redacted and either sender of
+ * mxEvent or has powerlevel sufficient to redact events other than their own.
+ * @param {MatrixEvent} mxEvent The event to test permission for
+ * @param {string} userId The user ID of the user to test permission for
+ * @return {boolean} true if the given used ID can redact given event
+ */
+RoomState.prototype.maySendRedactionForEvent = function(mxEvent, userId) {
+    const member = this.getMember(userId);
+    if (!member || member.membership === 'leave') return false;
+
+    if (mxEvent.status || mxEvent.isRedacted()) return false;
+    if (mxEvent.getSender() === userId) return true;
+
+    return this._hasSufficientPowerLevelFor('redact', userId);
+};
+
+/**
+ * Returns true if the given user ID has sufficient power level for type
+ * @param {string} type The type of power level to check against
+ * @param {string} userId The user ID of the user to test permission for
+ * @return {boolean} true if the given user ID has sufficient power level
+ */
+RoomState.prototype._hasSufficientPowerLevelFor = function(type, userId) {
```

Comment:
`maySendEvent` checks inside `power_levels.events.X` yet this checks `power_levels.X`
Can't check whether we can redact using `maySendEvent`